### ISSUE TITLE
Annotation of long quotes is brittle

### ISSUE THREAD

**Issue by Vinnl** [tosdr/phoenix#618]
I'm not quite sure what the issue is, but my intuition is that there might be a different in how Javascript calculates string length and how Ruby does it - see e.g. https://mathiasbynens.be/notes/javascript-unicode#counting-symbols 

See the screencast below: I select a range of text up to and including the word `GDPR`, but once annotated, that word is no longer included in the snippet. From a frustrating debugging session, I know at least that the document below contains some non-breaking spaces - I'm not sure how many code units that symbol is.

![bug](https://user-images.githubusercontent.com/4251/43774172-914b8814-9a48-11e8-8c4b-524e14b24557.gif)

I'm not quite sure what the best way to deal with this. 

- We might consider laying out the snippets in Javascript rather than in Ruby, but then we also need to do the same everywhere a snippet appears, e.g. on the point's page.
- Another option is to somehow try to match the Javascript selection in Ruby against the document's text, but since the selection differs from the respective quote in the document (newlines in the document are spaces in the selection, for example), this can be difficult.
- We could also try to do some normalisation when crawling a document. This could include removing at least special kinds of whitespace, maybe even special characters in general (but I can imagine we also want to support e.g. Japanese ToS), removing tags, potentially replacing them with simple newlines, etc.

And that's even assuming that the issue here indeed is string length.

**Comment by michielbdejong** [comment]
I don't see the `br` and `strong` tags when testing this locally, but I guess that's because you made that screencast before you merged that PR.

When I try to annotate that same text ('ever .. GDPR'), it creates a point with a quoteStart of -1. I think that may be related to the tags removal? Will add some debug statements to investigate further.

**Comment by michielbdejong** [comment]
It is as I thought, now you cannot even select the short text `'ever.   Updated'` because in the database that text appears as:

```sh
phoenix_development=# select SUBSTRING(text, 212, 30) from documents where id=248;
    substring    
-----------------
 ever. </p>     +
 <p>            +
 <strong>Updated
(1 row)
```

but it tries to update the point to quoteStart -1 and quoteEnd 14:
```sh
"document_id"=>"248", "quoteRev"=>"latest", "quoteStart"=>"-1", "quoteEnd"=>"14", "quoteCaseId"=>"", "point_id"=>"4511", "id"=>"528"
```

**Comment by michielbdejong** [comment]
Will dig into the client-side javascript that results in those values -1 and 14.

**Comment by michielbdejong** [comment]
Commenting out the style for `.hiddenTag` fixes the problem for the quote `'ever.  <p> <strong>Updated</strong></p>'`. Will now investigate the problem with the longer `'ever ... GDPR'` quote.

**Comment by michielbdejong** [comment]
ok, fixed by that same hotfix. will merge that now; @Vinnl we can improve further when you're back.

**Comment by michielbdejong** [comment]
Fixed in #634
